arviz/plots/lmplot.py

```diff
@@ -0,0 +1,323 @@
+
+    x_var_names = None
+    if isinstance(x, str):
+        x = idata.constant_data[x]
+        x_skip_dims = x.dims
+    elif isinstance(x, tuple):
+        x_var_names = x
+        x = idata.constant_data
+        x_skip_dims = x.dims
+    elif x is None:
+        if plot_dim is None:
+            x = y.coords[y.dims[0]]
+        else:
+            x = y.coords[plot_dim]
+        x_skip_dims = x.dims
+    elif isinstance(x, DataArray):
+        x_skip_dims = x.dims
+    else:
+        x = xr.DataArray(x)
+        x_skip_dims = [x.dims[-1]]
+
+    if isinstance(y_model, str):
+        if "posterior" not in idata.groups():
+            warnings.warn("Posterior not found in idata", UserWarning)
+            y_model = None
+        elif hasattr(idata.posterior, y_model):
+            y_model = idata.posterior[y_model]
+        else:
+            warnings.warn("y_model not found in posterior", UserWarning)
+            y_model = None
+
+    if isinstance(y_hat, str):
+        if "posterior_predictive" not in idata.groups():
+            warnings.warn("posterior_predictive not found in idata", UserWarning)
+            y_hat = None
+        elif hasattr(idata.posterior_predictive, y_hat):
+            y_hat = idata.posterior_predictive[y_hat]
+        else:
+            warnings.warn("y_hat not found in posterior_predictive", UserWarning)
+            y_hat = None
+
+    pp_sample_ix = None
+    if (y_hat is not None and kind_pp == "samples") or (
+        y_model is not None and kind_model == "lines"
+    ):
+        if y_hat is not None:
+            total_pp_samples = y_hat.sizes["chain"] * y_hat.sizes["draw"]
+        else:
+            total_pp_samples = y_model.sizes["chain"] * y_model.sizes["draw"]
+
+        if (
+            not isinstance(num_pp_samples, Integral)
+            or num_pp_samples < 1
+            or num_pp_samples > total_pp_samples
+        ):
+            raise TypeError(
+                "`num_pp_samples` must be an integer between 1 and "
+                + "{limit}.".format(limit=total_pp_samples)
+            )
+
+        pp_sample_ix = np.random.choice(total_pp_samples, size=num_pp_samples, replace=False)
+
+    if plot_dim is None:
+        skip_dims = list(y.dims)
+    elif isinstance(plot_dim, str):
+        skip_dims = [plot_dim]
+    elif isinstance(plot_dim, tuple):
+        skip_dims = list(plot_dim)
+
+    x = filter_plotters_list(
+        list(
+            xarray_var_iter(
+                x,
+                var_names=x_var_names,
+                skip_dims=set(x_skip_dims),
+                combined=True,
+            )
+        ),
+        "plot_lm",
+    )
+
+    y = filter_plotters_list(
+        list(
+            xarray_var_iter(
+                y,
+                skip_dims=set(skip_dims),
+                combined=True,
+            )
+        ),
+        "plot_lm",
+    )
+
+    # If there are multiple x and multidimensional y, we need total of len(x)*len(y) graphs
```

Comment:
Thanks for the comment!

Reply:
Add more comments like this in various places. Its very helpful to understand whats going on in the code!